### PR TITLE
[Maintenance] Remove php platform from composer

### PR DESCRIPTION
**Pull request by lchrusciel**
I've removed the platform from the composer to let flex detect the current environment based on used CLI PHP. In addition, I've added composer v2

Replacecment for #434

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
 
                 - echo "memory_limit=4096M" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
 
-                - composer self-update --1
+                - composer self-update --preview
```
Comment (Zales0123) [inline .travis.yml:42]: `composer self-update --2`?

File: src/Migrations/Version20200916093101.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20200916093101 extends AbstractMigration
```
Comment (Zales0123) [inline src/Migrations/Version20200916093101.php:10]: Shoudn't it be loaded automatically from the `CoreBundle`?

File: .travis.yml (hunk 1 above)
Comment (lchrusciel) [reply .travis.yml:42]: I kept preview to have the newest versions, sometimes not stable even. We had it before v2 was announced, but I don't have a strong opinion about it. But if we will change it, I would go with `--stable`

File: src/Migrations/Version20200916093101.php (hunk 2 above)
Comment (lchrusciel) [reply src/Migrations/Version20200916093101.php:10]: It is opened to 1.7, where we don't have new migrations yet. During upmerge it should be removed

Comment (Zales0123) [reply src/Migrations/Version20200916093101.php:10]: Indeed 👍

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, Łukasz! :1st_place_medal:

**Pull request merged**